pkg/insights/metrics.go

```diff
@@ -0,0 +1,37 @@
+package insights
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	insightsMetricsRegistry *prometheus.Registry
+)
+
+func init() {
+	insightsMetricsRegistry = prometheus.NewRegistry()
+	MustRegisterMetricCollectors(
+		collectors.NewGoCollector(),
+		collectors.NewProcessCollector(collectors.ProcessCollectorOpts{}),
+	)
+	startMetricsServer()
+}
+
+// startMetricsServer starts an HTTP server for the Insights metrics registry.
+func startMetricsServer() {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.HandlerFor(insightsMetricsRegistry, promhttp.HandlerOpts{}))
+	go http.ListenAndServe(":8080", mux)
```

Comment:
I think it would be slightly better to move it to anonymous function and handle the error return value.

Reply:
That makes sense, but I'm wondering what should we do in case of an error. Kill the entire IO or just restart the server in a loop to not interfere with the rest of IO functionality?

Reply:
I am not sure if we need some extra logic. I would just start with logging the error.

Reply:
@tremes Okay, but what if it crashes? Will IO continue happily running without the metrics? That seems like it could lead to some problems, so I think having it in an infinite loop with a bit of a sleep in between restarts will be fine. That way, if something bad happens to the server, it will just restart and the metrics will be available again. Having IO run completely without metrics seems like it could cause trouble with debugging and the error message could get lost in the log quite easily. Having IO crash when the server crashes is a possibility, but that could totally break IO in case of some funny networking misconfiguration, possibly, which would then crashloop IO, and that doesn't sound very good to me, either.